swift/Sources/SignalClient/zkgroup/AuthCredentialPresentation.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright 2020-2021 Signal Messenger, LLC.
+// SPDX-License-Identifier: AGPL-3.0-only
+//
+
+import Foundation
+import SignalFfi
+
+public class AuthCredentialPresentation: ByteArray {
+
+  public static let SIZE: Int = 493
+
+  public required init(contents: [UInt8]) throws {
+    try super.init(newContents: contents, expectedLength: AuthCredentialPresentation.SIZE)
+
+    try withUnsafePointerToSerialized { contents in
+      try checkError(signal_auth_credential_presentation_check_valid_contents(contents))
+    }
+  }
+
+  public func getUuidCiphertext() throws -> UuidCiphertext {
+    return try withUnsafePointerToSerialized { contents in
+      try invokeFnReturningSerialized {
+        signal_auth_credential_presentation_get_uuid_ciphertext($0, contents)
+      }
+    }
+  }
+
+  public func getRedemptionTime() throws -> UInt32 {
```

Comment:
Nit: it would be nice to comment what the units of time are.

Reply:
Yeah, not a regression but definitely a worthwhile improvement.